src/js/reducers/layoutReducer.ts

```diff
@@ -0,0 +1,64 @@
+import {Map} from 'immutable';
+import {ActionType, getType} from 'typesafe-actions';
+import {LayoutState} from '../store/factory/Layout';
+import * as layoutActions from '../actions/layoutActions';
+
+/**
+ * This reducer handles layout updates
+ * @param {Object} state - An Immutable state object
+ * @param {Object} action - An action object
+ */
+export function layoutReducer(state: LayoutState,
+  action: ActionType<typeof layoutActions>): LayoutState {
+  // handle layout actions
+  const id = action.meta;
+
+  if (typeof state === 'undefined') {
+    return Map();
+  }
+
+  if (action.type === getType(layoutActions.baseAddLayout)) {
+    return state.set(String(id), action.payload);
+  }
+  else if (action.type === getType(layoutActions.baseAddGrouptoLayout)) {
```

Comment:
this is not a big deal, but stylistically and for consistency purposes i prefer just using separate if blocks instead of chaining `else`-if here because there's return statements at the bottom of each block. conceptually, reducers are more like switch-cases over the action types. it also helps you more easily skim the difference between `if`s that check the action types and if-else logic within each action handler